packages/relay/src/index.ts

```diff
@@ -116,45 +129,72 @@ class Relay implements IRelay {
       };
     }
 
-    return this.gossipSub.publish(this.pubSubTopic, msg);
+    return await this.gossipSub.publish(pubSubTopic, msg);
   }
 
-  /**
-   * Add an observer and associated Decoder to process incoming messages on a given content topic.
-   *
-   * @returns Function to delete the observer
-   */
   public subscribe<T extends IDecodedMessage>(
     decoders: IDecoder<T> | IDecoder<T>[],
     callback: Callback<T>
   ): () => void {
-    const contentTopicToObservers = Array.isArray(decoders)
-      ? toObservers(decoders, callback)
-      : toObservers([decoders], callback);
-
-    for (const contentTopic of contentTopicToObservers.keys()) {
-      const currObservers = this.observers.get(contentTopic) || new Set();
-      const newObservers =
-        contentTopicToObservers.get(contentTopic) || new Set();
+    const pubSubTopicToContentObservers = Array.isArray(decoders)
+      ? toObservers(Array.from(this.pubSubTopics), decoders, callback)
+      : toObservers(Array.from(this.pubSubTopics), [decoders], callback);
+
+    for (const [
+      pubSubTopic,
+      contentTopicToObservers
+    ] of pubSubTopicToContentObservers.entries()) {
+      const existingContentObservers =
+        this.observers.get(pubSubTopic) ||
+        new Map<ContentTopic, Set<Observer<T>>>();
+
+      for (const [
+        contentTopic,
+        newObservers
+      ] of contentTopicToObservers.entries()) {
+        const currObservers =
+          existingContentObservers.get(contentTopic) || new Set<Observer<T>>();
```

Comment:
```suggestion
          existingContentObservers.get(contentTopic) ?? new Set<Observer<T>>();
```

It is better practice to use the nullish coalescing operator so that the left value is never misinterpreted as unset when it's actually set with a value that coerces to `false`.

It does not apply in this specific case but it's just better practice in general.

Ref: https://dev.to/hereisnaman/logical-or-vs-nullish-coalescing-operator-in-javascript-3851